src/main/java/evolvio/render/Circle.java

```diff
@@ -0,0 +1,33 @@
+package evolvio.render;
+
+import java.awt.Color;
+
+public class Circle {
+    private final double x;
+    private final double y;
+    private final double radius;
+    private final Color color;
```

Comment:
Those values may have to change at runtime.
Keep in mind that creating new Objects is always slower than setting a value :smile:

Reply:
Making things immutable to begin with makes everything so much easier to reason about. If it were to be used on mobile where it would have noticeable effects on CPU and memory use then I'd agree. Creating simple objects like this has become a *lot* more efficient over the years so imho it really doesn't matter. Immediately offering setters just encourages bad habits too. If performance becomes an issue and making classes mutable improves it then fair enough.